### PR TITLE
[CALCITE-2279] Fix issue in DruidJsonFilter reported by errorprone

### PR DESCRIPTION
**Pull request by nishantmonu51**

### CODE REVIEW

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidJsonFilter.java [hunk 1]
```diff
@@ -321,7 +321,7 @@ private static DruidJsonFilter toBetweenDruidFilter(RexNode rexNode, RelDataType
       return null;
     }
     final boolean isNumeric = lhs.getType().getFamily() == SqlTypeFamily.NUMERIC
-        || lhs.getType().getFamily() == SqlTypeFamily.NUMERIC;
```
Comment (b-slim) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidJsonFilter.java]: @nishantmonu51 thanks can you please add a test case for this?

Comment (nishantmonu51) [reply druid/src/main/java/org/apache/calcite/adapter/druid/DruidJsonFilter.java]: done. added test @b-slim

**Comment by nishantmonu51** [comment]
+cc @jcamachor @b-slim : please review.

**Comment by b-slim** [comment]
@nishantmonu51 i don't see the tests, plus i see there is some code changes unrelated, not sure what is happening really

**Comment by zabetak** [comment]
I am closing this PR as the issue has been fixed in https://github.com/apache/calcite/commit/155276591288615c4d02d55fb7d77eceb2e24b2d.

**Pull request closed without merging**